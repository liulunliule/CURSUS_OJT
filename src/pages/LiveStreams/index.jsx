import React, { useEffect, useState } from "react";
import "./index.scss";
import { Link } from "react-router-dom";
import Stream from "../../assets/img/stream.png";
import { useSelector } from "react-redux";

const LiveStreams = () => {
    const [mouseOverColor, setMouseOverColor] = useState(null);

    const mouseOverOn = () => {
        setMouseOverColor("black");
    };

    const mouseOverOff = () => {
        setMouseOverColor(null);
    };

    const [data, setData] = useState([]);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        const fetchData = async () => {
            try {
                const response = await fetch("https://66908324c0a7969efd9c4d55.mockapi.io/liveInstructor");
                const result = await response.json();
                setData(result);
                setLoading(false);
            } catch (error) {
                console.error("Error fetching data:", error);
                setLoading(false);
            }
        };

        fetchData();
    }, []);

    const isShowAll = useSelector((state) => state.savedCourse.isShowAll);
    return (
        <div className={`wrapper12 ${isShowAll ? "active" : ""}`}>
            <div>
                <div className="container-fluid">
                    <div className="row">
                        <div className="right_side">
                            <div className="col-xl-9 col-lg-8">
                                <div className="section3125">
                                    <h4 className="item_title">
                                        All Live Streams
                                    </h4>
                                    <div className="la5lo2">
                                        <div className="row">
                                            {data.map((d) => (
                                                <div className="col-md-3">
                                                    <div className="stream_1 mb-4">
                                                        <Link
                                                            to="/view_live_streams"
                                                            className="stream_bg"
                                                        >
                                                            <img src={d.img} />
                                                            <h4>{d.name}</h4>
                                                            <p>
                                                                Live
                                                                <span></span>
                                                            </p>
                                                        </Link>
                                                    </div>
                                                </div>
                                            ))}

                                            <div className="col-md-12 mt-10">
                                                <div className="spinner">
                                                    <div className="bounce1"></div>
                                                    <div className="bounce2"></div>
                                                    <div className="bounce3"></div>
                                                </div>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                            </div>

                            <div className="col-xl-3 col-lg-4">
                                <div className="fcrse_11 mt-20">
                                    <div className="cater_ttle">
                                        <h4>Live Streaming</h4>
                                    </div>
                                    <div className="live_text">
                                        <div className="live_icon">
                                            <img src={Stream} />
                                        </div>
                                        <div class="live-content">
                                            <p>
                                                Set up your channel and stream
                                                live to your students
                                            </p>
                                            <Link to="/Add_Live_Stream">
                                                <button
                                                    style={{
                                                        backgroundColor:
                                                            mouseOverColor,
                                                    }}
                                                    class="live_link"
                                                    onMouseOver={mouseOverOn}
                                                    onMouseOut={mouseOverOff}
                                                >
                                                    Get Started
                                                </button>
                                            </Link>
                                            <span class="livinfo">
                                                Info : This feature only for
                                                'Instructors'.
                                            </span>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default LiveStreams;
