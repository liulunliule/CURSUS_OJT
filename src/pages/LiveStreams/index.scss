.wrapper {
    padding: 90px 0 0;
    transition: all 0.3s ease-out;
    margin-left: 240px;
    background-color: #F8F8F8;
    transition: margin-left 0.5s ease;

    &.active{
        transition: margin-left 0.5s ease;
        margin-left: 0;
    }
}

a{
    text-decoration: none;
}


.section3125 {
    float: left;
    width: 100%;
}

.item_title {
    font-size: 18px;
    font-weight: bold;
    font-family: 'Roboto', sans-serif;
    color: #333;
    text-align: left;
    float: left;
    margin-bottom: 20px;
}

.see150 {
    font-size: 14px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #afafaf;
    text-align: right;
    float: right;
    padding-top: 3px;
    margin-bottom: 20px;
}

.la5lo2 {
    float: left;
    width: 95%;
    margin-left: 34px;
}

.stream_1 {
    width: 100%;
    border-radius: 3px;
    padding: 8px;
    background-position: center;
    background-repeat: no-repeat;
    background: rgba(51,1,51, 0.1);
}

.stream_bg {
    text-align: center;
    display: inline-block;
    width: 100%;
    padding: 15px 10px;
    border-radius: 3px;
    background: rgba(255, 255, 255, 0.06);
    backdrop-filter: blur(30px);
    box-shadow: 0 6px 25px rgba(0, 0, 0, 0.13);
}

.stream_bg img {
    width: 60px !important;
    height: 60px !important;
    border-radius: 100%;
    display: inline-block !important;
    border: 1px solid #fff;
    box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.1);
}

.stream_bg h4 {
    margin-top: 10px;
    margin-bottom: 5px;
    font-size: 14px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    color: #333;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
}

.stream_bg p span {
    width: 6px;
    height: 6px;
    background: #ed2a26;
    position: absolute;
    z-index: 100;
    border-radius: 100%;
}

.right_side {
    margin-left: auto;
}

.tutor_img {
    width: 100%;
    float: left;
    padding: 20px;
    text-align: center;
}

.right_side {
    display: flex;
    // margin-left: 10px;
}

.cater_ttle {
    float: left;
    width: 100%;
    padding: 15px 20px;
    border-bottom: 1px solid #efefef;
}

.fcrse_11 {
    background: #fff;
    width: 88%;
    height: 360px;
    float: left;
    border-radius: 3px;
    border: 1px solid #efefef;
    transition: all .2s ease-in-out;
    margin-left: 20px;
    margin-right: 5px;
    margin-top: 20px;

}

.fcrse_11 h4 {
    font-size: 16px;
    color: #333;
    float: left;
    font-weight: 800;
    font-family: 'Roboto', sans-serif;
    margin-bottom: 0;
}

.live_text {
    float: left;
    width: 100%;
    text-align: center;
    padding: 20px;
}

.live_icon {
    float: left;
    width: 100%;
    text-align: center;
    margin-bottom: 20px;
}

.live_icon img {
    width: 80px;
    height: 80px;
    align-self: center;
}

.live-content {
    float: left;
    width: 100%;
    padding: 0;
    text-align: center;
}

.live-content p {
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    line-height: 26px;
    margin-bottom: 20px;
    color: #686f7a;
}

.live_link {
    height: 40px;
    background: #ed2a26;
    border: 0;
    border-radius: 3px;
    padding: 10px 20px;
    color: #fff;
    font-size: 14px;
}

.livinfo {
    font-size: 12px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    margin-bottom: 0;
    margin-top: 15px;
    color: #686f7a;
    display: block;
}

.spinner {
    margin: 0 auto 0;
    width: 50px;
    text-align: center;
    height: 20px;
    border-radius: 100px;
    background-color: #fff;
    line-height: 20px;
    border: 1px solid #efefef;
    cursor: pointer;
}

.spinner .bounce1 {
    -webkit-animation-delay: -0.32s;
    animation-delay: -0.32s;
    background-color: #333;
}

.spinner > div {
    width: 10px;
    height: 10px;
    border-radius: 100%;
    display: inline-block;
    -webkit-animation: sk-bouncedelay 1.4s infinite ease-in-out both;
    animation: sk-bouncedelay 1.4s infinite ease-in-out both;
}

.spinner .bounce2 {
    -webkit-animation-delay: -0.16s;
    animation-delay: -0.16s;
    background-color: #333;
    opacity: 0.702;
}

.spinner .bounce3 {
    -webkit-animation-delay: -0.12s;
    animation-delay: -0.12s;
    background-color: #333;
    opacity: 0.4;
}