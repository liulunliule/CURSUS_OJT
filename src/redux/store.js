import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
