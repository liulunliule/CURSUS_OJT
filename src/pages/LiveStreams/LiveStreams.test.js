import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import LiveStreams from '.';

test('test verification page', () => {
  const tree = renderer.create(
    <Provider store={store}>
        <MemoryRouter>
            <LiveStreams/>
        </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});