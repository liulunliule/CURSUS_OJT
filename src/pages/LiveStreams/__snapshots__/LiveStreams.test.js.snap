// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test verification page 1`] = `
<div
  className="wrapper12 "
>
  <div>
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="right_side"
        >
          <div
            className="col-xl-9 col-lg-8"
          >
            <div
              className="section3125"
            >
              <h4
                className="item_title"
              >
                All Live Streams
              </h4>
              <div
                className="la5lo2"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-md-12 mt-10"
                  >
                    <div
                      className="spinner"
                    >
                      <div
                        className="bounce1"
                      />
                      <div
                        className="bounce2"
                      />
                      <div
                        className="bounce3"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-xl-3 col-lg-4"
          >
            <div
              className="fcrse_11 mt-20"
            >
              <div
                className="cater_ttle"
              >
                <h4>
                  Live Streaming
                </h4>
              </div>
              <div
                className="live_text"
              >
                <div
                  className="live_icon"
                >
                  <img
                    src="stream.png"
                  />
                </div>
                <div
                  class="live-content"
                >
                  <p>
                    Set up your channel and stream live to your students
                  </p>
                  <a
                    href="/Add_Live_Stream"
                    onClick={[Function]}
                  >
                    <button
                      class="live_link"
                      onMouseOut={[Function]}
                      onMouseOver={[Function]}
                      style={
                        Object {
                          "backgroundColor": null,
                        }
                      }
                    >
                      Get Started
                    </button>
                  </a>
                  <span
                    class="livinfo"
                  >
                    Info : This feature only for 'Instructors'.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
